src/thermo/PengRobinson.cpp

```diff
@@ -0,0 +1,797 @@
+//! @file PengRobinson.cpp
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at https://cantera.org/license.txt for license and copyright information.
+
+#include "cantera/thermo/PengRobinson.h"
+#include "cantera/thermo/ThermoFactory.h"
+#include "cantera/thermo/Species.h"
+#include "cantera/base/stringUtils.h"
+#include "cantera/base/ctml.h"
+
+#include <boost/math/tools/roots.hpp>
+
+#define _USE_MATH_DEFINES
+#include <math.h>
+
+using namespace std;
+namespace bmt = boost::math::tools;
+
+namespace Cantera
+{
+
+const double PengRobinson::omega_a = 4.5723552892138218E-01;
+const double PengRobinson::omega_b = 7.77960739038885E-02;
+const double PengRobinson::omega_vc = 3.07401308698703833E-01;
+
+PengRobinson::PengRobinson(const std::string& infile, const std::string& id_) :
+    m_b(0.0),
+    m_a(0.0),
+    m_aAlpha_mix(0.0),
+    m_NSolns(0),
+    m_dpdV(0.0),
+    m_dpdT(0.0)
+{
+    fill_n(m_Vroot, 3, 0.0);
+    initThermoFile(infile, id_);
+}
+
+void PengRobinson::setSpeciesCoeffs(const std::string& species, double a, double b, double w)
+{   
+    size_t k = speciesIndex(species);
+    if (k == npos) {
+        throw CanteraError("PengRobinson::setSpeciesCoeffs",
+            "Unknown species '{}'.", species);
+    }
+
```

Comment:
According to the definition proposed by Pitzer, acentric factor is not likely to be less than -1. So is it more appropriate to detect the effectiveness of ω and report the incorrect input here?

Reply:
Hi, @yuji-1 - thanks for this comment!

There is always a balance in what the user is allowed to input, in terms of what is known/thought to be "correct," and what a user might want to try out, for some reason perhaps not known to us.  Think of it as a balance between "rules" and "freedom."

In this case, so long as an acentric factor less than -1 does not lead to errors or failures in the code, my preference would be to allow it.